apps/stable_diffusion/src/utils/stencils/zoe/__init__.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import torch
+from pathlib import Path
+import requests
+
+
+from einops import rearrange
+
+remote_model_path = (
+    "https://huggingface.co/lllyasviel/Annotators/resolve/main/ZoeD_M12_N.pt"
+)
+
+
+class ZoeDetector:
+    def __init__(self):
+        cwd = Path.cwd()
+        ckpt_path = Path(cwd, "stencil_annotator")
+        ckpt_path.mkdir(parents=True, exist_ok=True)
+        modelpath = ckpt_path / "ZoeD_M12_N.pt"
+
+        with requests.get(remote_model_path, stream=True) as r:
+            r.raise_for_status()
+            with open(modelpath, "wb") as f:
+                for chunk in r.iter_content(chunk_size=8192):
+                    f.write(chunk)
+
+        midas = torch.hub.load(
```

Comment:
- `AyaanShah2204/ZoeDepth` is configured to trigger a `torch.hub.load` for `AyaanShah2204/MiDaS`. (This happens in the `./zoedepth/models/base_models/midas.py` line 341) 
- `AyaanShah2204/MiDaS` has the necessary changes to not require a old versions of `timm`
- Theoretically, if we were to remove this line of code, it still would work and we would be able to unpin MiDaS

I don't have SHARK set up on my local machine so I have no way to verify this.